tests/filecheck/xdsl-opt.mlir

```diff
@@ -0,0 +1,32 @@
+// RUN: xdsl-opt -split-input-file %s | xdsl-opt -split-input-file | filecheck %s
+
+"builtin.module"() ({
+}) : () -> ()
+
+// -----
+"builtin.module"() ({
+  "test.op"() : () -> ()
+}) : () -> ()
+
+// -----
+"builtin.module"() ({
+  %x = "test.op"() : () -> i1
+}) : () -> ()
+
+// -----
+"builtin.module"() ({
+  %x = "test.op"() : () -> i2
+}) : () -> ()
+
+// CHECK: "builtin.module"() ({
+// CHECK: "test.op"() : () -> ()
+// CHECK: }) : () -> ()
+
+// CHECK: "builtin.module"() ({
+// CHECK:   %x = "test.op"() : () -> i1
```

Comment:
I think three more changes to make this perfect:

 - [ ] move the checks closer to what they're checking
 - [ ] use CHECK and CHECK-NEXT